src/nlp_types.jl

```diff
@@ -212,3 +215,50 @@ function print(io :: IO, nlp :: NLPModelMeta)
     @printf(io, "nonlinear network constraints:   "); dsp(nlp.nnet'); @printf(io, "\n");
   end
 end
+
+"""
+    bound_constrained(nlp)
+    bound_constrained(meta)
+
+Returns whether the problem has bounds on the variables.
+"""
+bound_constrained(meta::NLPModelMeta) = length(meta.ifree) < meta.nvar
+bound_constrained(nlp::AbstractNLPModel) = bound_constrained(nlp.meta)
```

Comment:
Usually when we say a problem is bound constrained, we mean that it has *only* bound constraints. Here, the problem could have other contraints, right?